flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusReporterTaskScopeTest.java

```diff
@@ -52,228 +43,159 @@
-                                "flink_taskmanager_job_task_my_gauge", LABEL_NAMES, labelValues1))
+                                getLogicalScope(METRIC_NAME), LABEL_NAMES, LABEL_VALUES_1))
                 .isEqualTo(3.);
         assertThat(
                         CollectorRegistry.defaultRegistry.getSampleValue(
-                                "flink_taskmanager_job_task_my_gauge", LABEL_NAMES, labelValues2))
+                                getLogicalScope(METRIC_NAME), LABEL_NAMES, LABEL_VALUES_2))
                 .isEqualTo(4.);
     }
 
     @Test
-    void metersCanBeAddedSeveralTimesIfTheyDifferInLabels() throws UnirestException {
-        Meter meter = new TestMeter();
+    void metersCanBeAddedSeveralTimesIfTheyDifferInLabels() {
+        Meter meter1 = new TestMeter(1, 1.0);
+        Meter meter2 = new TestMeter(2, 2.0);
 
-        taskMetricGroup1.meter("my_meter", meter);
-        taskMetricGroup2.meter("my_meter", meter);
+        reporter.notifyOfAddedMetric(meter1, METRIC_NAME, metricGroup1);
+        reporter.notifyOfAddedMetric(meter2, METRIC_NAME, metricGroup2);
 
         assertThat(
                         CollectorRegistry.defaultRegistry.getSampleValue(
-                                "flink_taskmanager_job_task_my_meter", LABEL_NAMES, labelValues1))
-                .isEqualTo(5.);
+                                getLogicalScope(METRIC_NAME), LABEL_NAMES, LABEL_VALUES_1))
+                .isEqualTo(meter1.getRate());
         assertThat(
                         CollectorRegistry.defaultRegistry.getSampleValue(
-                                "flink_taskmanager_job_task_my_meter", LABEL_NAMES, labelValues2))
-                .isEqualTo(5.);
+                                getLogicalScope(METRIC_NAME), LABEL_NAMES, LABEL_VALUES_2))
+                .isEqualTo(meter2.getRate());
     }
 
     @Test
     void histogramsCanBeAddedSeveralTimesIfTheyDifferInLabels() throws UnirestException {
-        Histogram histogram = new TestHistogram();
+        TestHistogram histogram1 = new TestHistogram();
+        histogram1.setCount(1);
+        TestHistogram histogram2 = new TestHistogram();
+        histogram2.setCount(2);
 
-        taskMetricGroup1.histogram("my_histogram", histogram);
-        taskMetricGroup2.histogram("my_histogram", histogram);
+        reporter.notifyOfAddedMetric(histogram1, METRIC_NAME, metricGroup1);
+        reporter.notifyOfAddedMetric(histogram2, METRIC_NAME, metricGroup2);
 
         final String exportedMetrics = pollMetrics(reporter.getPort()).getBody();
-        assertThat(exportedMetrics)
-                .contains("subtask_index=\"0\",quantile=\"0.5\",} 0.5"); // histogram
-        assertThat(exportedMetrics)
-                .contains("subtask_index=\"1\",quantile=\"0.5\",} 0.5"); // histogram
+        assertThat(exportedMetrics).contains("label2=\"value1_2\",} 1.0"); // histogram
+        assertThat(exportedMetrics).contains("label2=\"value2_2\",} 2.0"); // histogram
```

Comment:
```suggestion
        assertThat(exportedMetrics).contains("label2=\"value1_2\",} 1.0");
        assertThat(exportedMetrics).contains("label2=\"value2_2\",} 2.0");
```
nit: I know that the comments were also part of my proposal. But they really don't add any value...